### PR TITLE
MWPW-116458 - Merch at Scale

### PR DESCRIPTION
**Pull request by auniverseaway**
* New Merch block
* Add an offer id and you get a price
* Add a link and you get a checkout button
* Link / button can be localized
* Supports all M@S price templates: strikethrough, with tax, optical, etc.

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/cmillar/mas?martech=off
- After: https://milo.adobe.com/drafts/cmillar/mas?martech=off

### CODE REVIEW

**Comment by auniverseaway** [comment]
Screen Shot 2022-09-03 at 9 32 11 AM

**Comment by auniverseaway** [comment]
Screen Shot 2022-09-03 at 9 27 19 AM

This won't work until it's live on https://milo.adobe.com.

File: libs/blocks/mas/mas.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { loadScript, getConfig, createTag } from '../../utils/utils.js';
+
+const { miloLibs, codeRoot } = getConfig();
+const base = miloLibs || codeRoot;
+
+await loadScript(`${base}/deps/tacocat-index.js`);
+
+export default async function init(el) {
+  const offerDiv = el.querySelector(':scope > div:first-child > div');
```
Comment (rgclayton) [inline libs/blocks/mas/mas.js:9]: There is another pattern to display - a strike through price followed by a promo price.

File: libs/blocks/mas/mas.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const buttonDiv = el.querySelector(':scope > div:last-child > div');
+  const osi = offerDiv.textContent;
+  if (!osi) return;
```
Comment (rgclayton) [inline libs/blocks/mas/mas.js:12]: There is also a button only option.

**Review by yesil — APPROVED** [review]

File: libs/blocks/mas/mas.js (hunk 1 above)
Comment (auniverseaway) [reply libs/blocks/mas/mas.js:9]: Thanks. This isn't meant to be comprehensive by any means. I'll see if I can add this easily.

File: libs/blocks/mas/mas.js (hunk 2 above)
Comment (auniverseaway) [reply libs/blocks/mas/mas.js:12]: Same with the above. This is really a MVP to show folks you can do M@S on Milo.

**Comment by rgclayton** [comment]
Ok cool. Mainly I just wanted to note a few things that came to mind.

 You can assign to me and I can take care of these things.

**Comment by auniverseaway** [comment]
@rgclayton LOL. This is super low priority compared to Carousel. Just a fun late Friday project with friends.

**Review by rgclayton — APPROVED** [review]

**Pull request merged**